src/gpp_client/rest.py

```diff
@@ -0,0 +1,108 @@
+import aiohttp
+import certifi
+import gzip
+import ssl
+
```

Comment:
Let's add a `lock` since this will be used in high concurrency applications like GOATS and the Scheduler. 
`import asyncio`